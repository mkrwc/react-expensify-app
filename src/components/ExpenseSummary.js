import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = ({ expensesTotal, expenseCount }) => {
    const expenseTaxonomy = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral( expensesTotal/ 100).format('$0,0.00');

    return (
        <div>
            <p>Viewing {expenseCount} {expenseTaxonomy} totalling {formattedTotal} </p>
        </div>        
    );
}

const mapStateToProps = (state) => {
    const displayedExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount: displayedExpenses.length,
        expensesTotal: getExpensesTotal(displayedExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
