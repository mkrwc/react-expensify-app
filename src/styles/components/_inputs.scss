.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}

.select {
    border: 1px solid #cacccd;
    height: 50px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}

.textarea {
    border: 1px solid #cacccd;
    height: 10rem;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}