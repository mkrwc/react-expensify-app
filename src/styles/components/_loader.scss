.loader {
    height: 100vh;
    height: 100vh;
    align-items: center;
    display: flex;
    justify-content: center;
}

.loader__image {
    height: 6rem;
    width: 6rem;
}