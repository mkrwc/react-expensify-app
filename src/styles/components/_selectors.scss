.button {
    background: #1c88bf;
    border: none;
    color:white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--logout {
    background: none;
    border: none;
    color:white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button-secondary {
    @extend .button;
    background: #888
}