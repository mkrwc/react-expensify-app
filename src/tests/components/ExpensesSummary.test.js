import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with two expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesTotal={9434} expenseCount={2}/>);
    expect(wrapper).toMatchSnapshot();
});
