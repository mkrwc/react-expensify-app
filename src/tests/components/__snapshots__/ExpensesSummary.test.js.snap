// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with two expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totalling 
    $94.34
     
  </p>
</div>
`;
